From 46d90f92ca52f601122af8b1a4c3113c32c31732 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:44:06 -0400
Subject: [ethernet] cxgb4: Avoid dumping Write-only registers in register dump

Message-id: <1410475447-94392-108-git-send-email-svemuri@redhat.com>
Patchwork-id: 93863
O-Subject: [RHEL7.1 PATCH BZ 1124947 107/108] cxgb4: Avoid dumping Write-only registers in register dump
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Avoid dumping MPS_RPLC_MAP_CTL for reg dumps; this is a Write-Only register.
Reading this register may cause MPS TCAM corruption.

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 3d9103f80d7b1e8f0be85d87d102ba8dc1084f2d)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index 0393821..8c3b952 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -1604,7 +1604,8 @@ static void get_regs(struct net_device *dev, struct ethtool_regs *regs,
   0xd004, 0xd03c,
   0xdfc0, 0xdfe0,
   0xe000, 0xea7c,
-  0xf000, 0x11190,
+  0xf000, 0x11110,
+  0x11118, 0x11190,
   0x19040, 0x1906c,
   0x19078, 0x19080,
   0x1908c, 0x19124,
@@ -1811,7 +1812,8 @@ static void get_regs(struct net_device *dev, struct ethtool_regs *regs,
   0xd004, 0xd03c,
   0xdfc0, 0xdfe0,
   0xe000, 0x11088,
-  0x1109c, 0x1117c,
+  0x1109c, 0x11110,
+  0x11118, 0x1117c,
   0x11190, 0x11204,
   0x19040, 0x1906c,
   0x19078, 0x19080,
-- 
1.7.1